From b7026173f467b3ae7e56323bd21201ae69787939 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dennis=20Nienh=C3=BCser?= <earthwings@gentoo.org>
Date: Sat, 22 Dec 2012 04:36:47 -0500
Subject: [PATCH] Fix compilation after API changes between beta 1 and rc 1.

Please review/test.
CCMAIL: simon@simonzone.com
CCMAIL: dilfridge@gentoo.org
---
 src/bindings/python/sip/GeoDataGeometry.sip      |  2 +-
 src/bindings/python/sip/GeoDataMultiGeometry.sip |  2 +-
 src/bindings/python/sip/GeoDataPolygon.sip       | 26 +++---------------------
 src/bindings/python/sip/GeoDataRegion.sip        |  2 +-
 src/bindings/python/sip/GeoDataTrack.sip         |  2 +-
 src/bindings/python/sip/GeoDocument.sip          |  6 ------
 src/bindings/python/sip/GeoGraphicsItem.sip      |  3 ---
 src/bindings/python/sip/RenderPlugin.sip         |  9 --------
 8 files changed, 7 insertions(+), 45 deletions(-)

diff --git a/src/bindings/python/sip/GeoDataGeometry.sip b/src/bindings/python/sip/GeoDataGeometry.sip
index 28dbe63..7d88606 100644
--- a/src/bindings/python/sip/GeoDataGeometry.sip
+++ b/src/bindings/python/sip/GeoDataGeometry.sip
@@ -42,7 +42,7 @@ public:
 private:
 //ig                            GeoDataGeometry (GeoDataGeometryPrivate* priv);
 public:
-    virtual Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
+    virtual const Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
 };
 // GeoDataGeometry
 
diff --git a/src/bindings/python/sip/GeoDataMultiGeometry.sip b/src/bindings/python/sip/GeoDataMultiGeometry.sip
index 825ef48..9659fc7 100644
--- a/src/bindings/python/sip/GeoDataMultiGeometry.sip
+++ b/src/bindings/python/sip/GeoDataMultiGeometry.sip
@@ -49,7 +49,7 @@ public:
     virtual void            pack (QDataStream& stream) const;
     virtual void            unpack (QDataStream& stream);
     virtual ~GeoDataMultiGeometry ();
-    virtual Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
+    virtual const Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
     Marble::GeoDataGeometry*  child (int);
     int                     childPosition (Marble::GeoDataGeometry* child);
     void                    append (Marble::GeoDataGeometry* other);
diff --git a/src/bindings/python/sip/GeoDataPolygon.sip b/src/bindings/python/sip/GeoDataPolygon.sip
index aa2303c..ff0efe2 100644
--- a/src/bindings/python/sip/GeoDataPolygon.sip
+++ b/src/bindings/python/sip/GeoDataPolygon.sip
@@ -33,10 +33,10 @@ public:
     void                    setTessellate (bool tessellate);
     Marble::TessellationFlags  tessellationFlags () const;
     void                    setTessellationFlags (Marble::TessellationFlags f);
-    Marble::GeoDataLatLonAltBox  latLonAltBox () const;
-    Marble::GeoDataLinearRing&  outerBoundary () const;
+    virtual const Marble::GeoDataLatLonAltBox& latLonAltBox () const;
+    Marble::GeoDataLinearRing&  outerBoundary ();
     void                    setOuterBoundary (const Marble::GeoDataLinearRing& boundary);
-    QVector<Marble::GeoDataLinearRing>&  innerBoundaries () const;
+    QVector<Marble::GeoDataLinearRing>&  innerBoundaries ();
     void                    appendInnerBoundary (const Marble::GeoDataLinearRing& boundary);
     virtual void            pack (QDataStream& stream) const;
     virtual void            unpack (QDataStream& stream);
@@ -48,26 +48,6 @@ public:
 };
 // GeoDataPolygon
 
-
-class GeoDataOuterBoundary : Marble::GeoDataPolygon
-{
-%TypeHeaderCode
-#include <GeoDataPolygon.h>
-%End
-
-};
-// GeoDataOuterBoundary
-
-
-class GeoDataInnerBoundary : Marble::GeoDataPolygon
-{
-%TypeHeaderCode
-#include <GeoDataPolygon.h>
-%End
-
-};
-// GeoDataInnerBoundary
-
 };
 // Marble
 
diff --git a/src/bindings/python/sip/GeoDataRegion.sip b/src/bindings/python/sip/GeoDataRegion.sip
index 752fe69..98478fb 100644
--- a/src/bindings/python/sip/GeoDataRegion.sip
+++ b/src/bindings/python/sip/GeoDataRegion.sip
@@ -31,7 +31,7 @@ public:
     explicit                GeoDataRegion (Marble::GeoDataFeature* feature);
     virtual ~GeoDataRegion ();
     virtual const char*     nodeType () const;
-    Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
+    const Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
     void                    setLatLonAltBox (const Marble::GeoDataLatLonAltBox& latLonAltBox);
     Marble::GeoDataLod&     lod () const;
     void                    setLod (const Marble::GeoDataLod& lod);
diff --git a/src/bindings/python/sip/GeoDataTrack.sip b/src/bindings/python/sip/GeoDataTrack.sip
index e704343..26c4436 100644
--- a/src/bindings/python/sip/GeoDataTrack.sip
+++ b/src/bindings/python/sip/GeoDataTrack.sip
@@ -49,7 +49,7 @@ public:
 //ig    void                    setExtendedData (const GeoDataExtendedData& extendedData);
     virtual const char*     nodeType () const;
     virtual Marble::EnumGeometryId  geometryId () const;
-    virtual Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
+    virtual const Marble::GeoDataLatLonAltBox&  latLonAltBox () const;
     virtual void            pack (QDataStream& stream) const;
     virtual void            unpack (QDataStream& stream);
 };
diff --git a/src/bindings/python/sip/GeoDocument.sip b/src/bindings/python/sip/GeoDocument.sip
index e11a225..95e0d12 100644
--- a/src/bindings/python/sip/GeoDocument.sip
+++ b/src/bindings/python/sip/GeoDocument.sip
@@ -99,13 +99,7 @@ protected:
             else if (dynamic_cast<Marble::GeoDataPoint*>(sipCpp))
                 sipType = sipType_Marble_GeoDataPoint;
             else if (dynamic_cast<Marble::GeoDataPolygon*>(sipCpp))
-                {
                 sipType = sipType_Marble_GeoDataPolygon;
-                if (dynamic_cast<Marble::GeoDataInnerBoundary*>(sipCpp))
-                    sipType = sipType_Marble_GeoDataInnerBoundary;
-                else if (dynamic_cast<Marble::GeoDataOuterBoundary*>(sipCpp))
-                    sipType = sipType_Marble_GeoDataOuterBoundary;
-                }
             else if (dynamic_cast<Marble::GeoDataTrack*>(sipCpp))
                 sipType = sipType_Marble_GeoDataTrack;
             }
diff --git a/src/bindings/python/sip/GeoGraphicsItem.sip b/src/bindings/python/sip/GeoGraphicsItem.sip
index 5cc4b67..eff28d5 100644
--- a/src/bindings/python/sip/GeoGraphicsItem.sip
+++ b/src/bindings/python/sip/GeoGraphicsItem.sip
@@ -29,9 +29,6 @@ class GeoGraphicsItem
 
 public:
                             GeoGraphicsItem ();
-    Marble::GeoDataCoordinates  coordinate () const;
-    void                    setCoordinate (const Marble::GeoDataCoordinates& point);
-
 
     virtual ~GeoGraphicsItem ();
     enum GeoGraphicsItemFlag
diff --git a/src/bindings/python/sip/RenderPlugin.sip b/src/bindings/python/sip/RenderPlugin.sip
index a9654ec..7570fa2 100644
--- a/src/bindings/python/sip/RenderPlugin.sip
+++ b/src/bindings/python/sip/RenderPlugin.sip
@@ -29,8 +29,6 @@ class RenderPlugin : QObject, Marble::RenderPluginInterface /Abstract/
 
 public:
     QAction*                action () const;
-    void                    applyItemState ();
-    void                    retrieveItemState ();
     bool                    enabled () const;
     bool                    visible () const;
     void                    setEnabled (bool enabled);
@@ -41,12 +39,6 @@ protected:
     bool                    eventFilter (QObject*, QEvent*);
 
 public:
-    enum ItemDataRole
-    {
-        NameId,
-        ConfigurationDialogAvailable,
-        BackendTypes
-    };
     virtual ~RenderPlugin ();
     virtual QList<QActionGroup*>*  actionGroups () const;
     virtual QList<QActionGroup*>*  toolbarActionGroups () const;
@@ -62,7 +54,6 @@ public:
     };
     virtual Marble::RenderPlugin::RenderType  renderType () const;
     const Marble::MarbleModel*  marbleModel () const;
-    QStandardItem*          item ();
     void                    restoreDefaultSettings ();
 signals:
     void                    repaintNeeded (QRegion dirtyRegion = QRegion());
-- 
1.8.0.2

